libraries/mbed/targets/hal/TARGET_STM/TARGET_STM32F0/can_api.c

```diff
@@ -0,0 +1,458 @@
+            break;
+        case MODE_SILENT:
+            can->BTR |= CAN_BTR_SILM;
+            can->BTR &= ~CAN_BTR_LBKM;
+            success = 1;
+            break;
+        case MODE_TEST_GLOBAL:
+        case MODE_TEST_LOCAL:
+            can->BTR |= CAN_BTR_LBKM;
+            can->BTR &= ~CAN_BTR_SILM;
+            success = 1;
+            break;
+        case MODE_TEST_SILENT:
+            can->BTR |= (CAN_BTR_SILM | CAN_BTR_LBKM);
+            success = 1;
+            break;
+        default:
+            success = 0;
+            break;
+    }
+    can->MCR &= ~(uint32_t)CAN_MCR_INRQ;
+    while((can->MSR & CAN_MSR_INAK) == CAN_MSR_INAK) {
+    }
+    return success;
+}
+
+int can_filter(can_t *obj, uint32_t id, uint32_t mask, CANFormat format, int32_t handle) 
+{
+    CanHandle.Instance = (CAN_TypeDef *)(obj->can);
+    CAN_FilterConfTypeDef  sFilterConfig;
+    
+    sFilterConfig.FilterNumber = handle;
+    sFilterConfig.FilterMode = CAN_FILTERMODE_IDMASK;
+    sFilterConfig.FilterScale = CAN_FILTERSCALE_32BIT;
+    sFilterConfig.FilterIdHigh = (uint8_t) (id >> 8);
+    sFilterConfig.FilterIdLow = (uint8_t) id;
+    sFilterConfig.FilterMaskIdHigh = (uint8_t) (mask >> 8);
+    sFilterConfig.FilterMaskIdLow = (uint8_t) mask;
+    sFilterConfig.FilterFIFOAssignment = 0;
+    sFilterConfig.FilterActivation = ENABLE;
+    sFilterConfig.BankNumber = 14 + handle;
+  
+    HAL_CAN_ConfigFilter(&CanHandle, &sFilterConfig);
+      
+    return 0;
+}
+
+static void can_irq(CANName name, int id) 
+{
+    uint32_t tmp1 = 0, tmp2 = 0, tmp3 = 0;    
+    CanHandle.Instance = (CAN_TypeDef *)name;
+    
+    if(__HAL_CAN_GET_IT_SOURCE(&CanHandle, CAN_IT_TME)) {
+        tmp1 = __HAL_CAN_TRANSMIT_STATUS(&CanHandle, CAN_TXMAILBOX_0);
+        tmp2 = __HAL_CAN_TRANSMIT_STATUS(&CanHandle, CAN_TXMAILBOX_1);
+        tmp3 = __HAL_CAN_TRANSMIT_STATUS(&CanHandle, CAN_TXMAILBOX_2);
+        if(tmp1 || tmp2 || tmp3)  
+        {
+            irq_handler(can_irq_ids[id], IRQ_TX);
+        }
+    }
+  
+    tmp1 = __HAL_CAN_MSG_PENDING(&CanHandle, CAN_FIFO0);
+    tmp2 = __HAL_CAN_GET_IT_SOURCE(&CanHandle, CAN_IT_FMP0);
+  
+    if((tmp1 != 0) && tmp2) {
+         irq_handler(can_irq_ids[id], IRQ_RX);
+    }
+  
+    tmp1 = __HAL_CAN_GET_FLAG(&CanHandle, CAN_FLAG_EPV);
+    tmp2 = __HAL_CAN_GET_IT_SOURCE(&CanHandle, CAN_IT_EPV);
+    tmp3 = __HAL_CAN_GET_IT_SOURCE(&CanHandle, CAN_IT_ERR); 
+  
+    if(tmp1 && tmp2 && tmp3) {
+         irq_handler(can_irq_ids[id], IRQ_PASSIVE);
+    }
+  
+    tmp1 = __HAL_CAN_GET_FLAG(&CanHandle, CAN_FLAG_BOF);
+    tmp2 = __HAL_CAN_GET_IT_SOURCE(&CanHandle, CAN_IT_BOF);
+    tmp3 = __HAL_CAN_GET_IT_SOURCE(&CanHandle, CAN_IT_ERR);  
+    if(tmp1 && tmp2 && tmp3) {
+        irq_handler(can_irq_ids[id], IRQ_BUS);
+    }
+  
+    tmp3 = __HAL_CAN_GET_IT_SOURCE(&CanHandle, CAN_IT_ERR);  
+    if(tmp1 && tmp2 && tmp3) {
+        irq_handler(can_irq_ids[id], IRQ_ERROR);
+    }  
+}
+
+void CEC_CAN_IRQHandler(void)
```

Comment:
is this weak IRQ symbol? You should use your won irq handler, register it via ``NVIC_SetVector``